lib/backend/k8s/conversion/conversion.go

```diff
@@ -446,8 +446,14 @@ func (c Converter) k8sRuleToCalico(rPeers []networkingv1.NetworkPolicyPeer, rPor
 	// TODO: This currently creates a lot of rules by making every combination of from / ports
 	// into a rule.  We can combine these so that we don't need as many rules!
 	for _, port := range ports {
+		protocol, calicoPorts, err := c.k8sPortToCalicoFields(port)
+		if err != nil {
+			// If we fail to parse this port, skip it.
+			log.WithError(err).Warn("Failed to parse k8s port. Dropping rule.")
+			return []apiv3.Rule{}
```

Comment:
Take a look at what we do in v2.6: https://github.com/projectcalico/libcalico-go/pull/774/files

It's similar but we return an error back one more level. I think I prefer that approach (and makes the two consistent)